sdk/trace/src/main/java/io/opentelemetry/sdk/trace/config/TraceConfig.java

```diff
@@ -352,16 +350,26 @@ public Builder setTraceIdRatioBased(double samplerRatio) {
      */
     public TraceConfig build() {
       TraceConfig traceConfig = autoBuild();
-      Preconditions.checkArgument(
-          traceConfig.getMaxNumberOfAttributes() > 0, "maxNumberOfAttributes");
-      Preconditions.checkArgument(traceConfig.getMaxNumberOfEvents() > 0, "maxNumberOfEvents");
-      Preconditions.checkArgument(traceConfig.getMaxNumberOfLinks() > 0, "maxNumberOfLinks");
-      Preconditions.checkArgument(
-          traceConfig.getMaxNumberOfAttributesPerEvent() > 0, "maxNumberOfAttributesPerEvent");
-      Preconditions.checkArgument(
-          traceConfig.getMaxNumberOfAttributesPerLink() > 0, "maxNumberOfAttributesPerLink");
-      Preconditions.checkArgument(
-          traceConfig.getMaxLengthOfAttributeValues() >= -1, "maxLengthOfAttributeValues");
+      if (traceConfig.getMaxNumberOfAttributes() <= 0) {
```

Comment:
Didn't we have some `Utils.checkArgument` function?

Reply:
Yeah, the SDK uses API classes quite a bit (obviously because it has to implement it!). I think this is a fine usage, as long as it's not in a strictly "internal" package.